### ISSUE TITLE
Comments on "CubeBouncer revisited - setting up Mixed Reality for Quest 3 with MRTK3 and Unity 6"

### ISSUE THREAD

**Issue by LocalJoost** [LocalJoost/BlogComments#476]
Original article: https://localjoost.github.io/CubeBouncer-revisited--setting-up-Mixed-Reality-for-Quest-3-with-MRTK3-and-Unity-6/

**Comment by meta-meta** [comment]
Thank you so much for this guide and the starter repo, Joost! As a Quest/MRTK developer I've felt pretty lost this past year, waiting for the dust to settle and feeling unsure about the status and longevity of Quest support going forward. It's so encouraging to have some current touchpoint from a seasoned MRTK dev such as yourself.

I've got this setup, but I'm noticing that both in Link and in a standalone Quest3 build, I'm only seeing the fallback controller models. Furthermore, I can manipulate a cube with hands but not with controllers. Any ideas?

According to the last comment on this issue, fallback controller model is expected: https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/issues/752

I've added Unity gltFast and KTX support. Though now I'm vaguely recalling from my last attempt some months back that the specific KTX package might be hard coded in MRTK, preventing the Quest models from loading.
<img width="554" alt="image" src="https://github.com/user-attachments/assets/a71c63c7-c072-4a89-93a4-dc5a23ffe8b3" />
<img width="570" alt="image" src="https://github.com/user-attachments/assets/a733abaa-c9d0-44da-839e-aadbcecfe022" />
<img width="523" alt="image" src="https://github.com/user-attachments/assets/8801043b-9c5a-4cc0-8fa1-3d1d33f9f152" />

**Comment by LocalJoost** [comment]
Hi, the cubes are bare basic Unity objects, they don't have any MRTK3 behaviours on them. So I am not surprised you can't manipulate them with your controllers. In fact, you cannot really manipulate them with hands either, try to grab one or resize it - that will fail. I am just using my own hand smash service. You might try to add MRKT3 ObjectManipulators to them, maybe that will help a little. But the actual 'smashing away' will only work with hands. I am not so very sure about controller models, having 'grown up' as a HoloLens developer, I very much concentrate on hand tracking based manipulations, paired with hand menus for settings. I tend to use controllers in Quest only for making view captures - my controllers usually don't even have batteries in them :D

So I am afraid I can't help you there. But nice touching base with a fellow MRKT developer!

**Comment by meta-meta** [comment]
No, I mean that I added my own cube to a blank scene with an `ObjectManipulator`. I was able to move, rotate and scale it with hand tracking far pointers, but not with controllers. Thanks though, I'll reach out on the MRTK repo.

Your blog has been invaluable.

**Comment by LocalJoost** [comment]
@meta-meta I would have to try myself. I seem to recall it used to work, but now I am not so sure anymore. Have you tried to deploy the MRTK3 hand tracking example scene and see if that works? Also, you might want to play with these settings, assuming you haven't done so already
![image](https://github.com/user-attachments/assets/ed98abcd-1038-4765-a8a6-e162abedeec8)

**Comment by meta-meta** [comment]
Following up on the Quest controller visualization.

Here's where MRTK has hard-coded specific GLTF packages `com.atteneder.gltfast` and `com.atteneder.ktx`: https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/blob/e0dee855ceeff448b3245196b5cb6da804e326cd/org.mixedrealitytoolkit.input/MRTK.Input.asmdef#L32

This comes into play here: https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/blob/e0dee855ceeff448b3245196b5cb6da804e326cd/org.mixedrealitytoolkit.input/Visualizers/ControllerVisualizer/ControllerModelLoader.cs#L54

At this point, `MROPENXR_PRESENT` was the blocker. So I installed Mixed Reality OpenXR Plugin. But then I was getting warnings in project validator that this feature is deprecated on Android and to use Unity or Meta OpenXR plugin (which I am). It didn't work anyway. So it looks like MRTK does not support this out of the box currently. Meta has its own `OVRRuntimeControllerPrefab` for this purpose, and I'll see about shoehorning that into the MRTK XR Rig.
![Image](https://github.com/user-attachments/assets/47438d3c-e709-4411-b792-cd4bfc03d0d6)

![Image](https://github.com/user-attachments/assets/21fe8a78-339b-4222-b740-2b2ba32aabd2)

**Comment by meta-meta** [comment]
Following up on the controller unable to manipulate. It is due to this bug: https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/issues/885

In other words, looks like I'll have to switch to the XRI3 branch.

**Comment by zxzkf1992** [comment]
Hello, I'm encountering an issue with MRTK3 in Unity 6. In the Editor, when holding the Space key to summon the right hand, I can't control the forward/backward movement of the right hand using the mouse wheel, but this functionality works properly in Unity 2022.3 LTS. Have you experienced this situation before?

**Comment by meta-meta** [comment]
Hi @zxzkf1992 yes that sounds familiar. If you scroll A LOT, I think you'll see that it does move, just ever so slightly. There's a multiplier in the input mapping that is set an order of magnitude or so too low for some reason. I forget exactly where the setting is. Try input system mapping.

https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/issues/505

**Comment by zxzkf1992** [comment]
Hi @meta-meta thx for your reply, you are right, i adjusted the scale of the scroll operation in the MRTKInputSimulatorControls settings separately, and now it can move normally.
